pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -0,0 +1,650 @@
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.LabelApp: "istio-ingressgateway",
+								},
+							},
+						},
+					},
+				},
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									v1beta1constants.GardenRole: v1beta1constants.GardenRoleControlPlane,
+									v1beta1constants.LabelApp:   v1beta1constants.LabelKubernetes,
+									v1beta1constants.LabelRole:  v1beta1constants.LabelAPIServer,
+								},
+							},
+						},
+					},
+				},
+			},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+				networkingv1.PolicyTypeEgress,
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, v.client, deployment, func() error {
+		maxSurge := intstr.FromInt(100)
+		maxUnavailable := intstr.FromInt(0)
+		deployment.Labels = map[string]string{
+			v1beta1constants.GardenRole:                           v1beta1constants.GardenRoleControlPlane,
+			v1beta1constants.LabelApp:                             DeploymentName,
+			v1beta1constants.LabelNetworkPolicyFromShootAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+		}
+		deployment.Spec = appsv1.DeploymentSpec{
+			Replicas:             pointer.Int32Ptr(v.replicas),
+			RevisionHistoryLimit: pointer.Int32Ptr(1),
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{
+				v1beta1constants.LabelApp: DeploymentName,
+			}},
+			Strategy: appsv1.DeploymentStrategy{
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxUnavailable: &maxUnavailable,
+					MaxSurge:       &maxSurge,
+				},
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1constants.GardenRole:                          v1beta1constants.GardenRoleControlPlane,
+						v1beta1constants.LabelApp:                            DeploymentName,
+						v1beta1constants.LabelNetworkPolicyToShootNetworks:   v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToDNS:             v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToPrivateNetworks: v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyFromPrometheus:    v1beta1constants.LabelNetworkPolicyAllowed,
+					},
+					Annotations: map[string]string{
+						"checksum/secret-" + tlsAuthSecret.Name: v.secrets.TLSAuth.Checksum,
+						"checksum/secret-" + dhSecret.Name:      v.secrets.DiffieHellmanKey.Checksum,
+					},
+				},
+				Spec: corev1.PodSpec{
+					PriorityClassName: "system-cluster-critical",
+					DNSPolicy:         corev1.DNSDefault, // make sure to not use the coredns for DNS resolution.
```

Comment:
Can we disable the automatic service account mount? It's not needed as none of the containers in the Pod needs to talk to the kube-apsierver of the seed cluster.

Reply:
You are absolutely right. None of the two containers of the pod need to talk to the kube-apiserver.
However, I am not sure I understand what you suggest exactly. Do you propose to set `automountServiceAccountToken: false` for the pod?

Reply:
@ScheererJ I think yes, that's the proposal.

Reply:
Done with the latest commit.